e2e/discussions.spec.ts

```diff
@@ -0,0 +1,465 @@
+import { Page, expect, test } from '@playwright/test'
+import { testAuth } from './fixtures/authenticatedUserPages'
+import { entityUrlPathname } from './helpers/entities'
+import {
+  setupProjectWithPermissions,
+  teardownProjectsAndFileHandles,
+} from './helpers/setupTeardown'
+import { dismissAlert } from './helpers/testUser'
+import { Project } from './helpers/types'
+import { UserPrefixes, userConfigs } from './helpers/userConfig'
+
+const discussionActionIconClasses = {
+  PIN: '.fa-thumb-tack.imageButton',
+  EDIT: '.fa-pencil.imageButton',
+  DELETE: '.syn-trash-o.imageButton',
+  TOGGLE_FOLLOW: '.fa-eye.imageButton',
+  LINK: '.fa-link.imageButton',
+}
+const discussionThreadSelector = '.discussionThread:visible'
+const discussionReplySelector = '.discussionReply:visible'
+
+const getDefaultDiscussionPath = (projectId: string) => {
+  return `${entityUrlPathname(projectId)}/discussion/default`
+}
+
+const getThreadTextbox = (page: Page) => {
+  return page.locator('.markdownEditor').getByRole('textbox')
+}
+
+const expectDiscussionPageLoaded = async (page: Page, projectId: string) => {
+  await testAuth.step('Default discussion page has loaded', async () => {
+    await page.waitForURL(getDefaultDiscussionPath(projectId))
+    await expect(
+      page.getByRole('heading', { name: 'Discussion' }),
+    ).toBeVisible()
+    await expect(page.getByRole('button', { name: 'New Thread' })).toBeVisible()
+    await expect(
+      page.getByRole('button', { name: 'Discussion Tools' }),
+    ).toBeVisible()
+  })
+}
+
+const expectThreadTableLoaded = async (
+  page: Page,
+  title: string,
+  author: string,
+  replies: string,
+  views: string,
+) => {
+  await testAuth.step('Thread table info is loaded', async () => {
+    const threadCells = page
+      .getByRole('row')
+      .filter({ hasText: title })
+      .getByRole('cell')
+
+    await expect(threadCells.first(), 'Should have correct title').toHaveText(
+      title,
+    )
```

Comment:
Include a [custom expect message](https://playwright.dev/docs/test-assertions#custom-expect-message) to make it easier to understand a failure

Reply:
Nice!